Source/Canary/Mqtt5Canary/Mqtt5Canary.swift

```diff
@@ -124,65 +130,85 @@ class Mqtt5CanaryTestContext: @unchecked Sendable {
   /// Client Operations
   func mqtt5CanaryOperationStart(clientIndex: Int) async throws {
     let canaryClient = try getCanaryClient(clientIndex)
-    if !canaryClient.is_connected {
+    if await canaryClient.getConnected() == false {
       await statistic.incrementTotalOperation()
-      try canaryClient.client!.start()
+      if let _client = await canaryClient.client {
+        // Client already exists, just start it
+        try _client.start()
+      } else {
+        // Client does not exist, create a new one
+        throw CanaryTestError.InvalidArgument
```

Comment:
We currently only have one type of `CanaryTestError` of `InvalidArgument`. It might be useful to expand the enums so we throw an error that points to what exactly happened. e.g. Client index not found, InvalidOperation, various operational failures.